### ISSUE TITLE
VSCode started to freeze with Cmder integration

### ISSUE THREAD

**Issue by atiyar** [cmderdev/cmder#2147]
I integrated Cmder with the VSCode terminal a while back with the following settings -

`
    "terminal.integrated.shell.windows": "C:\\Windows\\System32\\cmd.exe",
     "terminal.integrated.shellArgs.windows": [
        "/k C:\\ProgramData\\Cmder\\vendor\\init.bat"
    ]
`

and everything was running smooth. But recently VSCode stared to freeze frequently and I met with a dialog box saying "This window is no longer responding" with three options "Reopen", "Keep Waiting" and "Close".
I've noticed it happens only after I kill the terminal, not when I open the terminal or not while the terminal is still open. Ass soon as I kill the terminal, VSCode freezes, I wait, and then the message box appears.
I've disabled the Cmder integration (basically, removed the second line from above) and the problem is gone.
I'm using VSCode v1.37.1 and Cmder v1.3.11

**Update** - tested with Cmder v1.3.12. The issue exists.

**Comment by joshpeng** [comment]
I'm still experiencing this on a brand new computer with fresh VSCode 1.41.1 and Cmder 1.3.13. My Windows 10 is version 1909.

**Comment by gksrb4** [comment]
I have frequently occurred this issue.  
This happens because cmder waits for confirmation to close. This setting can be turned off.

![image](https://user-images.githubusercontent.com/49257699/83347968-cb880400-a363-11ea-98a0-b1bf70f9abe2.png)

next version, It should be better to set default setting in vscode_init.cmd.

**Comment by atiyar** [comment]
@gksrb4 I just tested setting the "Confirm close" to Never and Automatic. The issue still exists, the editor freezes.

Tested with -
Cmder - v1.3.14
VS Code - 1.45.1
VS Code settings - exactly as described in [Seamless VS Code Integration](https://github.com/cmderdev/cmder/wiki/Seamless-VS-Code-Integration)
Windows 10 - 1909

**Comment by daxgames** [comment]
@atiyar @gksrb4 those settings have absolutely nothing to do with Visual Studio Code integration.  

Visual Studio Code with Cmder is achieved by `code.exe`running the Cmder init script with its default settings that is normally run by a task in Conemu.  

The settings you mention are Conemu settings this they have no effect since Conemu is not running.

**Comment by daxgames** [comment]
Also if this happens when you kill the Cmder terminal in vscode as stated above there is nothing we can do to fix that.  Something is going wrong with the way vscode is doing that.

The question I have is how are you killing the terminal?  Are you typing `exit` or clicking a button?

**Comment by atiyar** [comment]
@daxgames that was a very good question, thanks for that. I was always killing the terminal clicking a button that vscode has and the editor always freezes. But I just tested killing the terminal by typing `exit` and the editor didn't freeze.
I myself too thought it might be an issue on vscode's end. I will create a new issue there.

**Comment by gksrb4** [comment]
> @gksrb4 I just tested setting the "Confirm close" to Never and Automatic. The issue still exists, the editor freezes.
> 
> Tested with -
> Cmder - v1.3.14
> VS Code - 1.45.1
> VS Code settings for integration - exactly as described in [Seamless VS Code Integration](https://github.com/cmderdev/cmder/wiki/Seamless-VS-Code-Integration)
> Windows 10 - 1909

sorry, It's right. too few tests were done.
It is not a solution.
